src/components/cylc/SVGTask.vue

```diff
@@ -253,84 +246,81 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
-      type: Number,
-      default: 0.7
-    },
+const props = defineProps({
+  task: {
+    required: true
   },
-  methods: {
-    getRunningStyle () {
-      if (
-        this.task.state === TaskState.RUNNING.name &&
-        this.startTime &&
-        this.task.task?.meanElapsedTime
-      ) {
-        // job start time in ms
-        const startTime = Date.parse(this.startTime)
-        // current time in ms
-        const now = Date.now()
-        // job elapsed time in s
-        const elapsedTime = ((now - startTime) / 1000)
-        const ret = `
-          animation-name: c8-task-progress-animation;
-          animation-timing-function: steps(50);
-          animation-iteration-count: 1;
-          animation-duration: ${this.task.task.meanElapsedTime}s;
-          animation-delay: -${elapsedTime}s;
-          animation-fill-mode: forwards;
-        `
-        return ret.replace('\n', ' ')
-      }
-      return ''
-    },
-    getModiferTransform () {
-      // Returns the translation required to position the ".modifier" nicely in
-      // relation to the ".status".
-
-      // Both ".status" and ".modifier" are centered at (50, 50), we need to
-      // move ".modifier" up and to the left so that the two don't touch and
-      // have a sensible gap between them
-
-      // translation = -(
-      //   # (1) the x/y translation to the edge of ".modifier"
-      //   (
-      //     (.modifier.outline.width + .modifier.outline.stroke)
-      //     * modifierSize * sin(45)
-      //   )
-      //   # (2) the x/y translation to the edge of ".status"
-      //   (.status.outline.width + .status.outline.stroke) * sin(45)
-      // )
-      const translation = -(
-        // (1) the x/y translation to the edge of ".modifier"
-        (35.35 * this.modifierSize) +
-        // (2) the x/y translation to the edge of ".status"
-        42.42
-      )
-      return `
-        scale(${this.modifierSize}, ${this.modifierSize})
-        translate(${translation}, ${translation})
-      `
-    },
+  startTime: {
+    // The start time as an ISO8601 date-time string in expanded format
+    // e.g. 2022-10-26T13:43:45Z
+    // TODO: aim to remove this in due course
+    // (we should be able to obtain this directly from the task)
+    type: String,
+    required: false
+  },
+  modifierSize: {
+    // Scale the size of the task state modifier
+    type: Number,
+    default: 0.7
+  },
+})
+
+function getRunningStyle () {
+  if (
+    props.task.state === TaskState.RUNNING.name &&
+    props.startTime &&
+    props.task.task?.meanElapsedTime
+  ) {
+    // job start time in ms (UTC)
+    const startTime = Date.parse(props.startTime)
+    // current time in ms (UTC)
+    const now = Date.now()
+    // job elapsed time in ms
+    const elapsedTime = now - startTime
+    return {
+      animationDuration: `${props.task.task.meanElapsedTime}s`,
+      animationDelay: `-${elapsedTime}ms`,
+      animationFillMode: 'forwards',
```

Comment:
I tried that initially, but it's not about which direction the animation runs in, it's about what state the animation remains in after it's finished running. And because tasks that run for the first time don't have a `meanElapsedTime` the animation duration will be 0 and the animation will have effectively finished running from the get go, and so the progress indicator will show as complete immediately.